### PR TITLE
[UI] Pull Request Download diff Button

### LINKED ISSUE #8444: [Feature Request] At Pull Request: download patchfile [Button]
add a option to download the patch of the PullRequest

GitLab has a buton and Github does via link

For example:
-> https://patch-diff.githubusercontent.com/raw/alpinelinux/aports/pull/10887.patch
-> https://patch-diff.githubusercontent.com/raw/alpinelinux/aports/pull/10887.diff

### PR DESCRIPTION
**Pull request by 6543**
this closes https://github.com/go-gitea/gitea/issues/8444

Bildschirmfoto zu 2019-10-12 00-23-27

### CODE REVIEW

File: templates/repo/diff/options_dropdown.tmpl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class="ui dropdown tiny button">
+	{{.i18n.Tr "repo.diff.options_button"}}
+	<i class="dropdown icon"></i>
+	<div class="menu">
+		<a class="item tiny basic toggle button" data-target="#diff-files">{{.i18n.Tr "repo.diff.show_diff_stats"}}</a>
+		<a class="item" href="http://localhost:3000/test/hometest/pulls/15.patch" download="15.patch">{{.i18n.Tr "repo.diff.download_patch"}}</a>
```
Comment (6543) [inline templates/repo/diff/options_dropdown.tmpl]: i need to find the Variables to generate this link ...

**Review by zeripath — APPROVED** [review]

File: docs/content/doc/features/comparison.en-us.md [hunk 2]
```diff
@@ -105,6 +105,7 @@ _Symbols used in table:_
 | Revert specific commits or a merge request | [✘](https://github.com/go-gitea/gitea/issues/5158) | ✘ | ✓ | ✓ | ✓ | ✓ | ✘ |
 | Pull/Merge requests templates | ✓ | ✓ | ✓ | ✓ | ✓ | ✘ | ✘ |
 | Cherry-picking changes | [✘](https://github.com/go-gitea/gitea/issues/5158) | ✘ | ✘ | ✓ | ✓ | ✘ | ✘ |
+| Download Patch | [✓](https://github.com/go-gitea/gitea/issues/8444) | ? | ✓ | ✓ | ✓ | ? | ? |
```
Comment (lafriks) [inline docs/content/doc/features/comparison.en-us.md]:
```suggestion
| Download Patch | ✓ | ✘ | ✓ | ✓ | ✓ | [/](https://jira.atlassian.com/plugins/servlet/mobile#issue/BCLOUD-8323) | ✘ |
```

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by 6543** [comment]
@lafriks can you look at it if a missed something else?

File: templates/repo/diff/options_dropdown.tmpl [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<div class="ui dropdown tiny button">
+	{{.i18n.Tr "repo.diff.options_button"}}
+	<i class="dropdown icon"></i>
+	<div class="menu">
+		<a class="item tiny basic toggle button" data-target="#diff-files">{{.i18n.Tr "repo.diff.show_diff_stats"}}</a>
+		<a class="item" href="{{$.RepoLink}}/pulls/{{.Issue.Index}}.patch" download="15.patch">{{.i18n.Tr "repo.diff.download_patch"}}</a>
```
Comment (guillep2k) [inline templates/repo/diff/options_dropdown.tmpl]: What does 15 stand for here? (and the line below)

Comment (6543) [reply templates/repo/diff/options_dropdown.tmpl]: O you got me!

Comment (6543) [reply templates/repo/diff/options_dropdown.tmpl]: -> 281715d082a8493ca29e50b93dab46693375de27

Comment (6543) [reply templates/repo/diff/options_dropdown.tmpl]: @guillep2k was testing the html5 property "download"

File: docs/content/doc/features/comparison.en-us.md (hunk 2 above)
Comment (6543) [reply docs/content/doc/features/comparison.en-us.md]: @lafriks :eyes: ?

**Review by guillep2k — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
Make lgtm work

**Pull request merged**